packages/core/useContextMenu/index.ts

```diff
@@ -0,0 +1,139 @@
+   * enabled to the entire page if not specified.
+   *
+   * @default document
+   */
+  target?: MaybeComputedElementRef
+
+  /**
+   * Fires when the visibility of the context menu changes.
+   * @param visible
+   */
+  onVisibleChange?: (visible: boolean) => void
+}
+
+export interface UseContextMenuReturn {
+  /**
+   * Whether the context menu is visible.
+   */
+  visible: Ref<boolean>
+
+  /**
+   * The absolute position of the context menu to the viewport.
+   */
+  position: Ref<Position>
+
+  /**
+   * Hide the context menu.
+   */
+  hide: () => void
+
+  /**
+   * Show the context menu.
+   *
+   * @param [position] absolute position to the viewport, default to the previous position.
+   */
+  show: (position?: Position) => void
+}
+
+export function useContextMenu(MenuElement: MaybeComputedElementRef, options: UseContextMenuOptions = {}): UseContextMenuReturn {
+  const { hideOnClick = true, onVisibleChange = noop, target } = options
+
+  const visible = ref(false)
+  const position = ref<Position>({ x: 0, y: 0 })
+
+  const accessMenuElementIfExists = (fn: (el: HTMLElement | SVGElement) => void) => {
+    const el = unrefElement(MenuElement)
+    if (el)
+      fn(el)
+  }
+
+  const show = () => visible.value = true
+  const hide = () => visible.value = false
+
+  useEventListener('scroll', hide)
+  useEventListener('click', hide)
+  useEventListener('contextmenu', hide, { capture: true })
+
+  const contextMenuHandler = (e: MouseEvent) => {
+    e.preventDefault()
+    position.value = {
+      x: e.clientX,
+      y: e.clientY,
+    }
+    show()
+    // prevent other(parent) context menu from showing
+    e.stopPropagation()
+  }
+
+  if (target) {
+    // register on the target element
+    watch(
+      () => unrefElement(target),
+      el => useEventListener(el, 'contextmenu', contextMenuHandler))
+  }
+  else {
+    // or register on the entire page
+    useEventListener('contextmenu', contextMenuHandler)
+  }
+
+  watch(visible, onVisibleChange)
+  watch(
+    () => unrefElement(MenuElement),
+    (element) => {
+      // initialize
+      accessMenuElementIfExists((el) => {
+        el.style.position = 'fixed'
+        el.style.display = 'none'
+      })
+
+      useEventListener(element, 'click', (e) => {
+        if (!resolveUnref(hideOnClick)) {
+          // hide it only when clicking outside of the menu,
+          // so stopPropagation when clicking on itself.
+          e.stopPropagation()
+        }
+      })
+    })
+
+  // automatically show/hide the `MenuElement`
+  watch(visible, (visible) => {
+    accessMenuElementIfExists((el) => {
+      el.style.display = visible ? 'block' : 'none'
+    })
+  }, { immediate: true })
+
+  // automatically update the position of the `MenuElement`
+  watchEffect(() => {
+    accessMenuElementIfExists((el) => {
+      el.style.left = `${position.value.x}px`
+      el.style.top = `${position.value.y}px`
+    })
+  })
+
+  return {
+    visible,
+    position,
+
```

Comment:
Please remove useless newlines.

Reply:
Actually the newline is for separating the `state` and `method` since there might be a lot more 
 property to return. \
see [core/useManualRefHistory/index.ts#L201](https://github.com/vueuse/vueuse/blob/08cbd6d9f4b72954eca1865ef1f136f61daba711/packages/core/useManualRefHistory/index.ts#L201)

but it's okay to remove it as well, thanks for the suggestions!